packages/lib-classifier/src/components/Classifier/components/SubTaskPopup/SubTaskPopup.js

```diff
@@ -0,0 +1,185 @@
+    border: 1px solid ${props.theme.global.colors['light-1']}
+  `}
+`
+
+// Prevent tasks from screwing up presentation when container is resized.
+const TaskBox = styled(Box)`
+  overflow: auto;
+  cursor: auto;
+`
+
+function storeMapper (stores) {
+  const { activeStepTasks } = stores.classifierStore.workflowSteps
+  const drawingTasks = activeStepTasks.filter(task => task.type === 'drawing')
+  const activeDrawingTask = (drawingTasks.length > 0) ? drawingTasks[0] : {}
+  
+  const {
+    activeMark,
+    subTaskMarkBounds,
+    subTaskVisibility,
+    setSubTaskVisibility,
+  } = activeDrawingTask
+  
+  return {
+    activeMark,
+    subTaskMarkBounds,
+    subTaskVisibility,
+    setSubTaskVisibility,
+  }
+}
+
+class SubTaskPopup extends React.Component {
+  constructor () {
+    super()
+  }
+  
+  // TODO: Split render() into various asyncStates?
+
+  render () {
+    const {
+      activeMark,
+      subTaskMarkBounds,
+      subTaskVisibility,
+      setSubTaskVisibility,
+    } = this.props
+    const { svg } = this.context
+    
+    const ready = true // TODO: check with TaskArea/components/Tasks/Tasks.js
+    const tasks = (activeMark && activeMark.tasks) ? activeMark.tasks : []
+    
+    // Calculate default position
+    let x = 0, y = 0;
+    const svgBounds = svg && svg.getBoundingClientRect()
+
+    if (svgBounds && subTaskMarkBounds) {
+      const markX = subTaskMarkBounds.x || 0
+      const markY = subTaskMarkBounds.y || 0
+      const markWidth = subTaskMarkBounds.width || 0
+      const markHeight = subTaskMarkBounds.height || 0
+      
+      const svgX = svgBounds.x || 0
+      const svgY = svgBounds.y || 0
+      
+      x = markX + markWidth * 0.5 - svgX
+      y = markY + markHeight * 0.5 - svgY
+      
+    }
+    
+    const defaultPosition = { x, y }
+    
+    if (subTaskVisibility && tasks.length > 0) {
+      return (
+        <Rnd
+          key={activeMark.id}
+          minWidth={100}
+          minHeight={100}
+          default={defaultPosition}
+          cancel=".subtaskpopup-element-that-ignores-drag-actions"
+        >
+          <StyledContainer pad="xsmall" fill>
+            <Box>
+              <CloseButton
+                alignSelf='end'
+                onClick={() => setSubTaskVisibility(false)}
+              />
+            </Box>
+            
+            {tasks.map((task, index) => {
+              // classifications.addAnnotation(task, value) retrieves any existing task annotation from the store
+              // or creates a new one if one doesn't exist.
+              // The name is a bit confusing.
+              const annotation = activeMark.addAnnotation(task)
+              task.setAnnotation(annotation)
+              const TaskComponent = observer(taskRegistry.get(task.type).TaskComponent)
+              
+              if (annotation && TaskComponent) {
+                return (
+                  <TaskBox
+                    key={task.taskKey}
```

Comment:
I think this should be keyed by `annotation.id`.

Reply:
I've changed this to `annotation.id`, but note that this _masks_ the problem of Text Tasks overwriting one another. Now, it's become the "merry go round" model of the problem.

Reply:
It's something to do with there only ever being one task in the store, but it's handling the annotations for more than one drawn line. When we change lines, the stored annotation is briefly out of sync with the task component, I think.